woocommerce/woocommerce
# add: remote logger request uri sanitisation (JS)

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/50984.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JN site with this branch
3. Go to `Tools -> WCA Test Helper -> Remote Logging`
4. Enable remote logging
5. Click on JS `Simulate Core Exception`, navigate to a wc-admin url with some arbitrary query params such as: `/wp-admin/admin.php?page=wc-admin&key=very_secret_key`
7. Check the log server page for the log entry (See test plan in D155228-code to see how to check the log entry)
8. Confirm the log properties include `request_uri` and the non-whitelisted parameters have been masked
### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
